charts/alluxio/templates/config/alluxio-mount-conf.yaml

```diff
@@ -0,0 +1,101 @@
+# (the "License"). You may not use this work except in compliance with the License, which is
+# available at www.apache.org/licenses/LICENSE-2.0
+#
+# This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+# either express or implied, as more fully set forth in the License.
+#
+# See the NOTICE file distributed with this work for information regarding copyright ownership.
+#
+{{- if eq .Values.master.mountConfigStorage "configmap"}}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  annotations:
+    "helm.sh/hook": "pre-install"
+    "helm.sh/hook-delete-policy": before-hook-creation
+  name: {{ template "alluxio.fullname" .  }}-mount-config
+  labels:
+    name: {{ template "alluxio.fullname" .  }}-mount-config
+    app: {{ template "alluxio.name" . }}
+    chart: {{ template "alluxio.chart" . }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+  ownerReferences:
+  {{- if .Values.owner.enabled }}
+    - apiVersion: {{ .Values.owner.apiVersion }}
+      blockOwnerDeletion: {{ .Values.owner.blockOwnerDeletion }}
+      controller: {{ .Values.owner.controller }}
+      kind: {{ .Values.owner.kind }}
+      name: {{ .Values.owner.name }}
+      uid: {{ .Values.owner.uid }}
+  {{- end }}
+data:
+  mount.info: |
+    {{- range $index, $value :=.Values.master.nonNativeMounts }}
+    {{$value}}
+    {{- end}}
+  entrypoint.script: |
+    #!/usr/bin/env bash
+    export ALLUXIO_JAVA_OPTS=$(echo "$ALLUXIO_JAVA_OPTS" | sed "s|/etc/fluid/secrets\(/[-._a-zA-Z0-9]\+\)\{2\}|\`cat &\`|g" )
+    echo "replace $ALLUXIO_JAVA_OPTS"
+    /entrypoint.sh $@
+  mount.script: |
+    #!/usr/bin/env bash
+    # set the -e option, when some command return non zero, the shell will exit, no need to check the executed status.
+    set -xe
+
+    # check alluxio master ready
+    alluxio fsadmin report
+
+    base_path=$(cd `dirname $0`; pwd)
+    # alluxio fs mount output like below:
+    # https://mirrors.bit.edu.cn/apache/spark  on  /spark  (web, ...)
+    # /underFSStorage                          on  /       (local, ...)
+
+    # replace multiple space with single space
+    mounted=$(alluxio fs mount | sed "s/\s\+/ /g")
+
+    # get already mounted path in alluxio, ignore the root path '/'
+    # use grouping command to avoid grep returning non zero, which will cause shell exit.
+    mounted_dst=$(echo "$mounted" | cut -d " " -f 3 | { grep -x -v "/" || true; } )
+
+    # mount.info content like below:
+    # /spark https://mirrors.bit.edu.cn/apache/spark --option k=v --readonly
+    # get all needed mount path, ignore the root path '/', "grep -x" match the line
+    need_mount_dst=$(cat ${base_path}/mount.info | cut -d " " -f 1 | { grep -x -v "/" || true; })
+
+    # unmount, find already mounted in alluxio but not existed in mount.info
+    # -x match the line, -v find the unmatched, -F pattern
+    unmount_dst=$(echo "$mounted_dst" | { grep -x -v -F "$need_mount_dst" || true; })
```

Comment:
If there are multiple line in `mount.info` what will happen when we `grep -x -v -F "$need_mount_dst"`?

Reply:
Plus, shall we also check if the mount src changed?

Reply:
not match any of line .

Reply:
> Plus, shall we also check if the mount src changed?

I think it is a new feature,  as the  [AnalyzePathsDelta](https://github.com/fluid-cloudnative/fluid/blob/1d15d54cbabf61dd7f234212d5f8da14ebe35c9e/pkg/utils/dataset.go#L152) function only use alluxio mount path (not src path) to recgonize dataset mount differences.

Reply:
> > Plus, shall we also check if the mount src changed?
> 
> I think it is a new feature, as the [AnalyzePathsDelta](https://github.com/fluid-cloudnative/fluid/blob/1d15d54cbabf61dd7f234212d5f8da14ebe35c9e/pkg/utils/dataset.go#L152) function only use alluxio mount path (not src path) to recgonize dataset mount differences.

Thx. Now I think this is okay.